### PR TITLE
Improve message retention

### PR DESCRIPTION
**Pull request by johnwatson484**
https://eaflood.atlassian.net/browse/SFI-1509﻿

### CODE REVIEW

File: app/messaging/update-requests-awaiting-debt-data.js [hunk 1]
```diff
@@ -3,13 +3,17 @@ const { getHoldCategoryId } = require('../holds')
...
+    const error = new Error(`Payment request does not include debt data: ${paymentRequest.invoiceNumber}`)
+    error.category = 'validation'
+    throw error
   }
 
   const originalPaymentRequest = await db.paymentRequest.findOne({ where: { invoiceNumber: paymentRequest.invoiceNumber } })
 
   if (!originalPaymentRequest) {
-    throw new Error(`No payment request matching invoice number: ${paymentRequest.invoiceNumber}`)
+    const error = new Error(`No payment request matching invoice number: ${paymentRequest.invoiceNumber}`)
```
Comment (steve-dickinson) [inline app/messaging/update-requests-awaiting-debt-data.js:14]: does this need `cause`, just for consistency?

File: app/messaging/update-requests-awaiting-debt-data.js [hunk 2]
```diff
@@ -3,13 +3,17 @@ const { getHoldCategoryId } = require('../holds')
 
 const updateRequestsAwaitingDebtData = async (paymentRequest) => {
   if (!paymentRequest.debtType) {
-    throw new Error(`Payment request does not include debt data: ${paymentRequest.invoiceNumber}`)
+    const error = new Error(`Payment request does not include debt data: ${paymentRequest.invoiceNumber}`)
```
Comment (steve-dickinson) [inline app/messaging/update-requests-awaiting-debt-data.js:6]: does this need cause, just for consistency?

**Review by steve-dickinson — DISMISSED** [review]

File: app/messaging/update-requests-awaiting-debt-data.js (hunk 2 above)
Comment (johnwatson484) [reply app/messaging/update-requests-awaiting-debt-data.js:6]: cause is for building up a stack trace of errors so if one error causes another you can chain them. This is the first error thrown in this chain so there is no cause to attach.

File: app/messaging/update-requests-awaiting-debt-data.js (hunk 1 above)
Comment (johnwatson484) [reply app/messaging/update-requests-awaiting-debt-data.js:14]: As above, there is no cause for this one

**Review by steve-dickinson — APPROVED** [review]
LGTM

**Pull request merged**